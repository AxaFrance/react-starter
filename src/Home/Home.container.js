import React, { useContext } from 'react';
import { Context } from 'context/Context';
import { HomeComponent } from './Home';

const EnhancedHome = ({ location, title, subtitle }) => {
  const { context } = useContext(Context);
  return (
    <HomeComponent
      location={location}
      title={title}
      subtitle={subtitle}
      context={context}
    />
  );
};

export default EnhancedHome;
