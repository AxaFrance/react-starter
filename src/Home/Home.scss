@import '@axa-fr/react-toolkit-core/dist/assets/scss/core.scss';

.home {
  .af-title--color {
    color: $color-gray-1;
  }
}
