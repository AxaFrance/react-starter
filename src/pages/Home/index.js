export { HomeComponent } from './Home';
export { default } from './Home.container';
