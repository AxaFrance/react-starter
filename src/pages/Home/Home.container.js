import React, { useContext } from 'react';
import { Context } from 'context/Context';
import { HomeComponent } from './Home';

const EnhancedHome = ({ title, subtitle }) => {
  const { context } = useContext(Context);
  return <HomeComponent title={title} subtitle={subtitle} context={context} />;
};

export default EnhancedHome;
