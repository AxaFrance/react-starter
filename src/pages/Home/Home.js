import React from 'react';
import PropTypes from 'prop-types';

import './Home.scss';

export const HomeComponent = ({ title, subtitle }) => (
  <div className="home container">
    <h1 className="af-title--content">{title}</h1>
    <h2>{subtitle}</h2>
    <p>
      Donec sollicitudin molestie malesuada. Donec sollicitudin molestie
      malesuada. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor
      eget felis porttitor volutpat. Sed porttitor lectus nibh. Curabitur arcu
      erat, accumsan id imperdiet et, porttitor at sem. Vivamus magna justo,
      lacinia eget consectetur sed, convallis at tellus. Proin eget tortor
      risus. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      Pellentesque in ipsum id orci porta dapibus.
    </p>

    <p>
      Sed porttitor lectus nibh. Curabitur non nulla sit amet nisl tempus
      convallis quis ac lectus. Mauris blandit aliquet elit, eget tincidunt nibh
      pulvinar a. Sed porttitor lectus nibh. Quisque velit nisi, pretium ut
      lacinia in, elementum id enim. Vivamus magna justo, lacinia eget
      consectetur sed, convallis at tellus. Nulla quis lorem ut libero malesuada
      feugiat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus
      suscipit tortor eget felis porttitor volutpat. Sed porttitor lectus nibh.
    </p>
  </div>
);

const propTypes = {
  subtitle: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

HomeComponent.propTypes = propTypes;
