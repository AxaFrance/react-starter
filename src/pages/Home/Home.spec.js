import React from 'react';
import { render } from '@testing-library/react';
import { HomeComponent } from './Home';

it('1. Renders Home component without crashing', () => {
  const { asFragment } = render(
    <HomeComponent title="Accueil" subtitle="Sous titre" />
  );
  expect(asFragment()).toMatchSnapshot();
});
