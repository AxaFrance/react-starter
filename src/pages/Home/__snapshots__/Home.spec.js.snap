// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Renders Home component without crashing 1`] = `
<DocumentFragment>
  <div
    class="home container"
  >
    <h1
      class="af-title--content"
    >
      Accueil
    </h1>
    <h2>
      Sous titre
    </h2>
    <p>
      Donec sollicitudin molestie malesuada. Donec sollicitudin molestie malesuada. Donec rutrum congue leo eget malesuada. Vivamus suscipit tortor eget felis porttitor volutpat. Sed porttitor lectus nibh. Curabitur arcu erat, accumsan id imperdiet et, porttitor at sem. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Proin eget tortor risus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque in ipsum id orci porta dapibus.
    </p>
    <p>
      Sed porttitor lectus nibh. Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Mauris blandit aliquet elit, eget tincidunt nibh pulvinar a. Sed porttitor lectus nibh. Quisque velit nisi, pretium ut lacinia in, elementum id enim. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Nulla quis lorem ut libero malesuada feugiat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus suscipit tortor eget felis porttitor volutpat. Sed porttitor lectus nibh.
    </p>
  </div>
</DocumentFragment>
`;
